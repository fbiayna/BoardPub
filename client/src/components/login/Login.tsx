/* eslint-disable no-use-before-define */
import React from 'react'
import { ImageBackground, StyleSheet, Text, View, Image } from 'react-native'

const background = { uri: 'https://trello-attachments.s3.amazonaws.com/5ec457ae0da99b76dcdba46b/5ec457af3dca9f5428a9f179/b713ff3791a65ab0ca93ae3e5f76f476/menu-609364_1280.jpg' }
const logo = { uri: 'https://trello-attachments.s3.amazonaws.com/5ec457ae0da99b76dcdba46b/5ec457af3dca9f5428a9f179/263e7f5a2d4e919205f4745afbf1a032/logo-boardpub.png' }

export default function Login () {
  return (
    <View style={styles.container}>
      <ImageBackground source={background} style={styles.backimage}>
        <View style={styles.shadow}>
          <View style={styles.logo}>
            <Image source={logo} style={styles.image} />
          </View>
          <View style={styles.textSlogan}>
            <Text style={styles.text}>Las mejores promociones de tus locales hosteleros, en una sola app.</Text>
          </View>
          <View style={styles.textAction}>
            <Text style={styles.text}>¿Qué estás buscando?</Text>
          </View>
          <View style={styles.user}>
            <View style={styles.buttonUser}>
              <Text style={styles.textUser}>¡HE VENIDO A COMER/BEBER!</Text>
            </View>
          </View>
          <View style={styles.admin}>
            <View style={styles.buttonAdmin}>
              <Text style={styles.textAdmin}>¡QUIERO PROMOCIONAR MI LOCAL!</Text>
            </View>
          </View>
        </View>
      </ImageBackground>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center'
  },
  backimage: {
    opacity: 0.85,
    width: '100%',
    height: '100%',
    resizeMode: 'cover',
    justifyContent: 'center'
  },
  shadow: {
    width: '100%',
    height: '100%',
    opacity: 1,
    backgroundColor: '#000'
  },
  logo: {
    alignItems: 'center',
    justifyContent: 'center',
    flex: 2.5
  },
  image: {
    width: 320,
    height: 100
  },
  textSlogan: {
    flex: 2,
    justifyContent: 'center',
    alignItems: 'center'
  },
  textAction: {
    flex: 1,
    alignItems: 'center'
  },
  text: {
    width: 250,
    fontWeight: 'bold',
    alignItems: 'center',
    fontSize: 20,
    lineHeight: 30,
    textAlign: 'center',
    color: '#fff'
  },
  textSearch: {
    flex: 1,
    fontWeight: 'bold',
    fontSize: 25,
    lineHeight: 30,
    textAlign: 'center',
    color: '#fff'
  },
  user: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'center'
  },
  buttonUser: {
    width: 260,
    height: 60,
    backgroundColor: '#92000A',
    justifyContent: 'center',
    borderRadius: 50,
    elevation: 4
  },
  textUser: {
    fontSize: 14,
    fontWeight: 'bold',
    lineHeight: 22,
    textAlign: 'center',
    color: '#fff'
  },
  admin: {
    flex: 2,
    alignItems: 'center'
  },
  buttonAdmin: {
    width: 260,
    height: 60,
    backgroundColor: '#1565C0',
    justifyContent: 'center',
    borderRadius: 50,
    elevation: 4
  },
  textAdmin: {
    fontSize: 14,
    fontWeight: 'bold',
    lineHeight: 22,
    textAlign: 'center',
    color: '#fff'
  }
})
