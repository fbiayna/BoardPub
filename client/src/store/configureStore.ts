import { applyMiddleware, compose, createStore } from 'redux'
import thunk from 'redux-thunk'
import rootReducer from '../reducers/boardPubReducer'

export default function configureStore (initialState: {} | undefined) {
  return createStore(rootReducer, initialState, compose(applyMiddleware(thunk)))
}
