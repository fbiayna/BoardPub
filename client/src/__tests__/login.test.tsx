/* eslint-disable no-use-before-define */
import { Provider } from 'react-redux'
import configureStore from '../store/configureStore'
import React from 'react'
import renderer from 'react-test-renderer'
import Login from '../components/login/Login'

describe('Login Component', () => {
  test('should be defined', () => {
    const store = configureStore({})
    store.dispatch = jest.fn()

    const login = renderer.create(<Provider store={store} >
      <Login /></Provider>)

    expect(login).toMatchSnapshot()
  })
})
