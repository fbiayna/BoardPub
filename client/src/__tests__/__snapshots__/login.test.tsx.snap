// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should be defined 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "height": "100%",
        "justifyContent": "center",
        "opacity": 0.8,
        "resizeMode": "cover",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://trello-attachments.s3.amazonaws.com/5ec457ae0da99b76dcdba46b/5ec457af3dca9f5428a9f179/b713ff3791a65ab0ca93ae3e5f76f476/menu-609364_1280.jpg",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#000",
          "height": "100%",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 2.5,
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://trello-attachments.s3.amazonaws.com/5ec457ae0da99b76dcdba46b/5ec457af3dca9f5428a9f179/263e7f5a2d4e919205f4745afbf1a032/logo-boardpub.png",
            }
          }
          style={
            Object {
              "height": 100,
              "width": 320,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 2,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "alignItems": "center",
              "color": "#fff",
              "fontSize": 20,
              "fontWeight": "bold",
              "lineHeight": 30,
              "textAlign": "center",
              "width": 250,
            }
          }
        >
          Las mejores promociones de tus locales hosteleros, en una sola app.
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "alignItems": "center",
              "color": "#fff",
              "fontSize": 20,
              "fontWeight": "bold",
              "lineHeight": 30,
              "textAlign": "center",
              "width": 250,
            }
          }
        >
          ¿Qué estás buscando?
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#92000A",
              "borderRadius": 50,
              "elevation": 4,
              "height": 60,
              "justifyContent": "center",
              "width": 260,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 14,
                "fontWeight": "bold",
                "lineHeight": 22,
                "textAlign": "center",
              }
            }
          >
            ¡HE VENIDO A COMER/BEBER!
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 2,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#1565C0",
              "borderRadius": 50,
              "elevation": 4,
              "height": 60,
              "justifyContent": "center",
              "width": 260,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 14,
                "fontWeight": "bold",
                "lineHeight": 22,
                "textAlign": "center",
              }
            }
          >
            ¡QUIERO PROMOCIONAR MI LOCAL!
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
